model-engine/model_engine_server/domain/use_cases/llm_fine_tuning_use_cases.py

```diff
@@ -63,6 +64,28 @@ def are_dataset_headers_valid(file_location: str):
     return all(required_header in current_headers for required_header in REQUIRED_COLUMNS)
 
 
+def check_file_is_valid(file_name: Optional[str], file_type: str):
+    """
+    Ensure the file is valid with required columns 'prompt' and 'response', isn't malformatted, and exists.
+    file_type: 'training' or 'validation'
+    """
+    try:
+        if file_name is not None and not are_dataset_headers_valid(file_name):
+            raise InvalidRequestException(
+                f"Required column headers {','.join(REQUIRED_COLUMNS)} not found in training dataset"
+            )
+        else:
+            return True
+    except FileNotFoundError:
+        raise InvalidRequestException(
+            f"The {file_type} file that you provided cannot be found. Please check that the filepath is correct."
+        )
+    except csv.Error as exc:
+        raise InvalidRequestException(
+            f"An error occurred while processing the {file_type} CSV file. Details: {exc}"
```

Comment:
nit: change message to "Cannot parse the {file_type} dataset as CSV. Details: {exc}"